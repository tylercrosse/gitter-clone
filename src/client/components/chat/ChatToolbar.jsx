import React from 'react';

export const ChatToolbar = () => (
  <aside className="chat-toolbar">
    o
  </aside>
);

export default ChatToolbar;
