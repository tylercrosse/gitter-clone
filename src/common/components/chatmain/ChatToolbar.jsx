import React from 'react';
import PropTypes from 'prop-types';
import _uniqBy from 'lodash/uniqBy';
import Avatar from '../Avatar';

export const ChatToolbar = ({ messages, user }) => {
  const currentUser = user.name ? [{...user}] : [];
  const allUsers = messages
    .map((message) => message.User)
    .concat(currentUser);
  const uniqueUsers = _uniqBy(allUsers, 'id');
  const avatars = uniqueUsers
    .map((singleUser) => (
      <li className="avatar" key={singleUser.id}>
        <Avatar user={singleUser} />
        {/* <div className="status" /> */}
      </li>
    ));
  return (
    <aside className="chat-toolbar">
      <ul className="roster">
        {avatars}
      </ul>
    </aside>
  );
};

ChatToolbar.propTypes = {
  messages: PropTypes.array.isRequired,
  user: PropTypes.object.isRequired,
};

export default ChatToolbar;
