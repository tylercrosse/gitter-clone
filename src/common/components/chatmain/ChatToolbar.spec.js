import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChatToolbar from './ChatToolbar';

const setup = (propOverrides) => {
  const props = Object.assign({
    messages: [
      {User: {name: 'Joe', id: 123}},
      {User: {name: 'Josh', id: 654}},
    ],
    user: {name: 'Liz', id: 323}
  }, propOverrides);

  const wrapper = shallow(<ChatToolbar {...props} />);

  return {
    props,
    wrapper
  };
};

describe('<ChatToolbar />', () => {
  it('should render correctly when logged in', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when not logged in', () => {
    const { wrapper } = setup({user: {}});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
