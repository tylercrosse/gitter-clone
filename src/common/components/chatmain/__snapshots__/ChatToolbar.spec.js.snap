// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatToolbar /> should render correctly when logged in 1`] = `
<aside
  className="chat-toolbar"
>
  <ul
    className="roster"
  >
    <li
      className="avatar"
    >
      <Connect(Avatar)
        user={
          Object {
            "id": 123,
            "name": "Joe",
          }
        }
      />
    </li>
    <li
      className="avatar"
    >
      <Connect(Avatar)
        user={
          Object {
            "id": 654,
            "name": "Josh",
          }
        }
      />
    </li>
    <li
      className="avatar"
    >
      <Connect(Avatar)
        user={
          Object {
            "id": 323,
            "name": "Liz",
          }
        }
      />
    </li>
  </ul>
</aside>
`;

exports[`<ChatToolbar /> should render correctly when not logged in 1`] = `
<aside
  className="chat-toolbar"
>
  <ul
    className="roster"
  >
    <li
      className="avatar"
    >
      <Connect(Avatar)
        user={
          Object {
            "id": 123,
            "name": "Joe",
          }
        }
      />
    </li>
    <li
      className="avatar"
    >
      <Connect(Avatar)
        user={
          Object {
            "id": 654,
            "name": "Josh",
          }
        }
      />
    </li>
  </ul>
</aside>
`;
